sql/backfill.go

```diff
@@ -89,46 +89,23 @@ func convertBackfillError(
 	return convertBatchError(tableDesc, *b, pErr)
 }
 
-var descriptorChangedVersionError = roachpb.NewErrorf("table descriptor has changed version")
-
-// getTableDescAtVersion attempts to read a descriptor from the database at a
-// specific version. It returns descriptorChangedVersionError if the current
-// version is not the passed in version.
-func getTableDescAtVersion(
-	txn *client.Txn, id ID, version DescriptorVersion,
-) (*TableDescriptor, *roachpb.Error) {
-	tableDesc, pErr := getTableDescFromID(txn, id)
+func (sc *SchemaChanger) getNonDeletedTableDesc(txn *client.Txn) (
+	*TableDescriptor, *roachpb.Error,
+) {
+	tableDesc, pErr := getTableDescFromID(txn, sc.tableID)
 	if pErr != nil {
 		return nil, pErr
 	}
-	if tableDesc == nil {
-		return nil, roachpb.NewError(&roachpb.DescriptorDeletedError{})
-	}
-	if version != tableDesc.Version {
-		return nil, descriptorChangedVersionError
+	if tableDesc.Deleted {
+		return nil, roachpb.NewError(
+			&roachpb.DescriptorNotFoundError{DescriptorId: uint32(sc.tableID)},
```

Comment:
I think this error should be swallowed by all callers. So my usual nit applies: this function could return `nil`.

In fact, I wonder if we wouldn't need this function at all if we used leases (regular leases, not schema change leases). So the backfill would take and release a lease for for every span. In the common, non-contended case, this wouldn't cost anything. If the table has been marked deleted, the leasing system will know and not hand you a new lease (you might need a bit of care to make that you're releasing leases to the `LeaseManager` if going through the planner's leasing routines, since the planner also caches leases... I think, not sure).